### ISSUE TITLE
Version 1.12.0+ no longer runs at all without any GPG shadowed private keys

### ISSUE THREAD

**Issue by enticedwanderer** [maximbaz/yubikey-touch-detector#60]
After upgrading from 1.11.0 it seems that the touch detector no longer starts at all.  I'm currently on 1.12.1 and it exits immediately.

```
> yubikey-touch-detector --stdout -v
DEBU[2024-11-29T21:10:00-05:00] Starting YubiKey touch detector
DEBU[2024-11-29T21:10:00-05:00] HMAC watcher on '/dev' is successfully established
DEBU[2024-11-29T21:10:00-05:00] U2F watcher on '/dev' is successfully established
DEBU[2024-11-29T21:10:00-05:00] No shadowed private keys found.
>
```

I'm using a solo key (which doesn't have a functional gpg integration like yubi key does). After taking a quick look I saw a recent change that returns out of main when any of the GPG validity checks fail, circumventing the all the other detectors in the process.

See: [main.go#L94](https://github.com/maximbaz/yubikey-touch-detector/blob/862214fb4cfdf7466d8bdff4c265804e8c1d4e7d/main.go#L94)

Probably, meant to just use skip GPG detector instead.

**Comment by enticedwanderer** [comment]
Another minor note is that this is somewhat related to [allowing configurable detectors in issue 43](https://github.com/maximbaz/yubikey-touch-detector/issues/43). Had that been in place, there would be a work around when something like this happens. Configurable flag to skip GPG for example.

I'll revert for now

**Comment by maximbaz** [comment]
Thanks for the report, should be fixed in `1.12.2`! :+1: 

> Had that been in place, there would be a work around when something like this happens. Configurable flag to skip GPG for example.

Hehe you underestimate my ability to break code - on the contrary, the more code there is, the more changes some of it would be broken at any point of time :stuck_out_tongue_winking_eye: And even if it did work, you'd have less incentive to report it :grin: Jokes aside, I aim to provide a good experience out of the box, rather than a bunch of configs to disable bad experiences.
